examples/vault-cluster-authentication-gce/README.md

```diff
@@ -1,36 +1,173 @@
+Vault. With one or more policies on hand, you can then finally create the authentication role.
+
+When you create a Role in Vault, you define the Policies that are attached to that
+Role, how principals who assume that Role will authenticate and other parameters
+related to the authentication of that role such as when does the token issued by
+a successful attempt will expire. When your Role uses the GCE GCP Auth method,
+you also specify which of the GCE properties will be required by the principal
+(in this case, the GCE Instance) in order to successfully authenticate.
+
+In our example we create a simple Vault Policy that allows writing and reading from
+secrets in the path `secret` namespaced with the prefix `example_`, and then create
+a Vault Role that allows authentication from all instances in a specific Zone and
+with certain labels. You can read more about Role creation and check which other
+instance metadata you can use on auth [here][create_role].
+
+
+```bash
+vault auth enable gcp
+
+vault policy write "example-policy" -<<EOF
+path "secret/example_*" {
+  capabilities = ["create", "read"]
+}
+EOF
+
+vault write \
+  auth/gcp/role/example_role_name \
+  project_id="<project id>" \
+  type="gce" \
+  policies="example-policy" \
+  bound_zones="asia-east1-a" \
+  bound_labels="example_label:example_value,another_label:another_value"
+```
+
+See the whole example script at [startup-script-vault.sh][startup_vault].
+
+
+### Authenticating from an instance
+
+The token used to authenticate to Vault is a [JSON Web Token (JWT)][jwt] that can
+be fetched on the GCE's instance metadata endpoint and will be part of the body of
+data sent with the login request.
+
+```bash
+JWT_TOKEN=$(curl \
+  --fail \
+  --header "Metadata-Flavor: Google" \
+  --get \
+  --data-urlencode "audience=vault/<role name>}" \
+  --data-urlencode "format=full" \
+  "http://metadata/computeMetadata/v1/instance/service-accounts/<service account email>/identity")
+
+LOGIN_PAYLOAD=$(cat <<EOF
+{
+  "role":"${example_role_name}",
+  "jwt":"$JWT_TOKEN"
+}
+EOF
+)
+curl --fail --request POST --data '$LOGIN_PAYLOAD' https://vault.service.consul:8200/v1/auth/gcp/login
+```
+
+After sending the login request to Vault, Vault will verify it against GCP and
+return a JSON object with your login information. This JSON contains the `client_token`,
+that you will send with your future operations requests to Vault.
+
+Please note, that this could also have been achieved using the Vault cli tool
+instead of using `curl`. To see the full script for authenticating check the
+[client startup script][startup_client].
 
-To see how to connect to the Vault cluster, initialize it, and start reading and writing secrets, head over to the
-[How do you use the Vault cluster?](https://github.com/hashicorp/terraform-google-vault/tree/master/modules/vault-cluster#how-do-you-use-the-vault-cluster) docs.
+[auth_diagram]: https://www.vaultproject.io/img/vault-gcp-gce-auth-workflow.svg
```

Comment:
We may want to copy this (and any other) images to the repo. Vault docs are updated often and there's no guarantee the URL won't change or be removed.